Source/Xamarin/Prism.DryIoc.Forms/PrismApplication.cs

```diff
@@ -0,0 +1,135 @@
+        /// <see cref="CreateContainerRules" />
+        /// </summary>
+        /// <returns>An instance of <see cref="IContainer" /></returns>
+        protected virtual IContainer CreateContainer()
+        {
+            var rules = CreateContainerRules();
+            return new Container(rules);
+        }
+
+        /// <summary>
+        /// Create <see cref="Rules" /> to alter behavior of <see cref="IContainer" />
+        /// </summary>
+        /// <remarks>
+        /// Default rule is to consult <see cref="Xamarin.Forms.DependencyService" /> if the requested type cannot be inferred from
+        /// <see cref="Container" />
+        /// </remarks>
+        /// <returns>An instance of <see cref="Rules" /></returns>
+        protected virtual Rules CreateContainerRules()
+        {
+            return UnknownServiceResolverRule.DependencyServiceResolverRule;
+        }
+
+        protected virtual void ConfigureContainer()
+        {
+            Container.RegisterInstance(Logger);
+            Container.RegisterInstance(ModuleCatalog);
+            Container.RegisterInstance(Container);
+            Container.Register<INavigationService, DryIocPageNavigationService>(
+                serviceKey: _navigationServiceKey,
+                setup: Setup.With(allowDisposableTransient: true));
+            Container.Register<IApplicationProvider, ApplicationProvider>();
+            Container.Register<IModuleManager, ModuleManager>(Reuse.Singleton);
+            Container.Register<IModuleInitializer, DryIocModuleInitializer>(Reuse.Singleton);
+            Container.Register<IEventAggregator, EventAggregator>(Reuse.Singleton);
+            Container.Register<IDependencyService, DependencyService>(Reuse.Singleton);
+            Container.Register<IPageDialogService, PageDialogService>(Reuse.Singleton);
+        }
+
+        protected override void InitializeModules()
+        {
+            if (ModuleCatalog.Modules.Any())
+            {
+                var manager = Container.Resolve<IModuleManager>();
+                manager.Run();
+            }
+        }
+
+        /// <summary>
+        /// Resolve <see cref="INavigationService"/>
+        /// </summary>
+        /// <remarks>
+        /// The <see cref="_navigationServiceKey"/> is used as service key when resolving
+        /// </remarks>
+        /// <returns>Instance of <see cref="INavigationService"/></returns>
+        protected override INavigationService CreateNavigationService()
+        {
+            return Container.Resolve<INavigationService>(_navigationServiceKey);
+        }
+
+        protected override void ConfigureViewModelLocator()
+        {
+            ViewModelLocationProvider.SetDefaultViewModelFactory((view, type) =>
+            {
+                var page = view as Page;
+                if (page != null)
+                {
+                    var navigationService = CreateNavigationService();
```

Comment:
Perhaps refactor this to `protected virtual object ResolveViewModelForView(object,Type)`.